partiql-ast/src/main/java/org/partiql/ast/literal/LiteralApprox.java

```diff
@@ -0,0 +1,52 @@
+package org.partiql.ast.literal;
+
+import lombok.EqualsAndHashCode;
+import org.jetbrains.annotations.NotNull;
+
+import java.math.BigDecimal;
+
+/**
+ * TODO docs
+ */
+@EqualsAndHashCode(callSuper = false)
+public class LiteralApprox extends Literal {
```

Comment:
I had originally thought this as well. There are some edge cases here. For example, `123` and `123.` cannot both be represented by a BigDecimal alone since BigDecimal doesn't preserve whether the `123` mantissa is specified with or without a decimal point.

We could provide some boolean field to distinguish between the two (e.g. `hasPeriod`); however I came to release that approximate numeric also has this same issue and may need that field as well. Example: `123e0` vs `123.e0`. To exactly preserve the text, we would even need to distinguish whether the approximate numeric's exponent had an implicit plus (`123e0`), explicit plus (`123e+0`), or explicit minus (`123e-0`).

This led me to think perhaps we should just be passing around strings for the AST representation of numerics, leaving the parser to perform simple validation and type tagging.